ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/ImmutablePhoto.java

```diff
@@ -0,0 +1,236 @@
+package net.gini.android.vision.internal.camera.photo;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import net.gini.android.vision.document.ImageDocument;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * @exclude
+ */
+class ImmutablePhoto implements Photo {
```

Comment:
Since gif and png cannot be edited, but we still show them in the Review Screen I refactored `Photo` to be an interface and created the `ImmutablePhoto` for showing pngs and gifs.

There is also a `MutablePhoto` which contains most of the logic from the old `Photo` class.